features/dd-sdk-android-flags-openfeature/src/test/kotlin/com/datadog/android/flags/openfeature/DatadogFlagsProviderTest.kt

```diff
@@ -26,17 +39,402 @@ import org.mockito.quality.Strictness
+    fun `M return string value W getStringEvaluation() {successful resolution}`(
+        forge: Forge,
+        @StringForgery flagKey: String,
+        @StringForgery expectedValue: String,
+        @StringForgery defaultValue: String,
+        @StringForgery variant: String
+    ) {
+        // Given
+        val resolution = ResolutionDetails(
+            value = expectedValue,
+            variant = variant,
+            reason = forge.aValueFrom(ResolutionReason::class.java)
+        )
+        whenever(mockFlagsClient.resolve(flagKey, defaultValue)).thenReturn(resolution)
+
+        // When
+        val result = provider.getStringEvaluation(flagKey, defaultValue, null)
+
+        // Then
+        assertThat(result.value).isEqualTo(expectedValue)
+        assertThat(result.variant).isEqualTo(variant)
+        assertThat(result.reason).isEqualTo(resolution.reason?.name)
+    }
+
+    // endregion
+
+    // region Integer Evaluation
+
+    @Test
+    fun `M return integer value W getIntegerEvaluation() {successful resolution}`(
+        forge: Forge,
+        @StringForgery flagKey: String
+    ) {
+        // Given
+        val expectedValue = forge.anInt()
+        val defaultValue = forge.anInt()
+        val resolution = ResolutionDetails(
+            value = expectedValue,
+            reason = forge.aValueFrom(ResolutionReason::class.java)
+        )
+        whenever(mockFlagsClient.resolve(flagKey, defaultValue)).thenReturn(resolution)
+
+        // When
+        val result = provider.getIntegerEvaluation(flagKey, defaultValue, null)
+
+        // Then
+        assertThat(result.value).isEqualTo(expectedValue)
+    }
+
+    // endregion
+
+    // region Double Evaluation
+
+    @Test
+    fun `M return double value W getDoubleEvaluation() {successful resolution}`(
+        forge: Forge,
+        @StringForgery flagKey: String
+    ) {
+        // Given
+        val expectedValue = forge.aDouble()
+        val defaultValue = forge.aDouble()
+        val resolution = ResolutionDetails(
+            value = expectedValue,
+            reason = forge.aValueFrom(ResolutionReason::class.java)
+        )
+        whenever(mockFlagsClient.resolve(flagKey, defaultValue)).thenReturn(resolution)
+
+        // When
+        val result = provider.getDoubleEvaluation(flagKey, defaultValue, null)
+
+        // Then
+        assertThat(result.value).isEqualTo(expectedValue)
+    }
+
+    // endregion
+
+    // region Object Evaluation
+
+    @Test
+    fun `M return Value structure W getObjectEvaluation() {successful resolution}`(
+        forge: Forge,
+        @StringForgery flagKey: String,
+        @StringForgery jsonKey: String,
+        @StringForgery jsonValue: String
+    ) {
+        // Given
+        val count = forge.anInt(min = 0, max = 1000)
+        val defaultValue = Value.Structure(mapOf("key" to Value.String(forge.anAlphabeticalString())))
+        val expectedJsonValue = JSONObject(mapOf(jsonKey to jsonValue, "count" to count))
```

Comment:
```suggestion
        val defaultValue = Value.Structure(forge.aMap { anAlphabeticalString() to Value.String(anAlphabeticalString()) })
        val expectedJsonValue = JSONObject(forge.aMap { anAlphabeticalString() to anAlphabeticalString() })
```

but ideally it is better to generate a map with different value types